File: frontend/src/microservices/advertising/objects/GenerateAdvertisementTextCandidatesResponse.ts
// 本文件定义 advertising 模块的 `GenerateAdvertisementTextCandidatesResponse`，作为文案候选响应并提供 JSON 编解码。

import type { AdvertisementTextCandidateResponse } from './AdvertisementTextCandidateResponse'

export type GenerateAdvertisementTextCandidatesResponse = {
  candidates: AdvertisementTextCandidateResponse[]
}

export const generateAdvertisementTextCandidatesResponseFromJson = (json: string): GenerateAdvertisementTextCandidatesResponse =>
  JSON.parse(json) as GenerateAdvertisementTextCandidatesResponse

export const generateAdvertisementTextCandidatesResponseToJson = (value: GenerateAdvertisementTextCandidatesResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/GenerateAdvertisementTextCandidatesResponse.scala
// 本文件定义 advertising 模块的 GenerateAdvertisementTextCandidatesResponse，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class GenerateAdvertisementTextCandidatesResponse(
    candidates: List[AdvertisementTextCandidateResponse]
)

object GenerateAdvertisementTextCandidatesResponse:
  given sourceEncoder: Encoder[GenerateAdvertisementTextCandidatesResponse] = deriveEncoder
  given sourceDecoder: Decoder[GenerateAdvertisementTextCandidatesResponse] = deriveDecoder

